components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/GetAuthenticatedApplicationsFunction.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.user;
+
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+
+import java.util.List;
+
+/**
+ * Function definition for retrieving authenticated applications.
+ */
+@FunctionalInterface
+public interface GetAuthenticatedApplicationsFunction {
+
+    /**
+     * Retrieve the already authenticated applications for a given session.
+     *
+     * @param context context object.
+     * @return List of already authenticated applications' name of the given session.
+     */
+    List<String> getAuthenticatedApplications(JsAuthenticationContext context);
```

Comment:
Other than just returning the authenticated application name, can we return a list of application objects? https://is.docs.wso2.com/en/latest/references/adaptive-authentication-js-api-reference/#application-object

Reply:
addressed with [https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/103/commits/cdc5fefbb62b736cc0bbafe4578b76050afea395](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/103/commits/cdc5fefbb62b736cc0bbafe4578b76050afea395)